### PR TITLE
extends filters to all equipments type

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/components/generic-filter.js [hunk 1]
```diff
@@ -54,13 +54,199 @@ const equipmentsDefinition = {
...
+            },
+            substationName: {
+                name: 'substationName',
+                type: filteredTypes.string,
+            },
+        },
+    },
 };
 
+function deepCopy(aObject) {
```
Comment (sBouzols) [inline src/components/generic-filter.js:251]: Strange util function here. Maybe we could use the spread operator [es6 spread operator](https://www.javatpoint.com/es6-spread-operator)
It works with arrays and objects

File: src/components/custom-list-item.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ export const CustomListItem = ({
             button
             key={item.name}
             selected={selected}
-            onClick={() => handleItemClicked(item)}
+            onMouseDown={() => handleItemClicked(item)}
```
Comment (sBouzols) [inline src/components/custom-list-item.js:118]: Why this change ? is this related to filter extension ?

**Review by sBouzols — COMMENTED** [review]
Some comments. Maybe another review by veteran contributor is necessary.

File: src/components/generic-filter.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/generic-filter.js:251]: Here, we cannot use spread operator, because we really need a deep copy.
This is a correction of an existing bug in the master, when we define a line filter and specify at the same time both nominal voltage 1 and nominal voltage 2 criteria : the nominal voltage 1 criteria values are overriden by the nominal voltage 2 criteria values ...

File: src/components/custom-list-item.js (hunk 2 above)
Comment (FranckLecuyer) [reply src/components/custom-list-item.js:118]: This is a correction of an existing bug in the master, when we delete an existing filter ... it tries to get the filter again, after it's deletion ...

**Review by niconoir — APPROVED** [review]

**Review by niconoir — APPROVED** [review]
